src/main/java/org/example/daos/DatabaseConnector.java

```diff
@@ -4,10 +4,9 @@
 import java.sql.DriverManager;
 import java.sql.SQLException;
 
-public final class DatabaseConnector {
+public class DatabaseConnector {
     private static Connection conn;
-    private DatabaseConnector() { }
-    public static Connection getConnection() throws SQLException {
+    public Connection getConnection() throws SQLException {
```

Comment:
any reason you made this non-static? a static method allows us to share the connection across multiple threads

Reply:
Unit tests arent running when it is static and i was unsure how else to fix it

getting this error message:

org.mockito.exceptions.misusing.MissingMethodInvocationException:
when() requires an argument which has to be 'a method call on a mock'.
For example:
when(mock.getArticles()).thenReturn(articles);

Also, this error might show up because:
1. you stub either of: final/private/equals()/hashCode() methods.
   Those methods *cannot* be stubbed/verified.
   Mocking methods declared on non-public parent classes is not supported.
2. inside when() you don't call method on mock but on some other object.